### PR TITLE
Renaming local goal_state_variable, adding tests and nit refactoring.

### PR DESCRIPTION
**Pull request by vrdmr**
## Description

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1691)

### CODE REVIEW

File: tests/data/wire/ext_conf.xml [hunk 1]
```diff
@@ -22,5 +22,5 @@
     <RuntimeSettings seqNo="0">{"runtimeSettings":[{"handlerSettings":{"protectedSettingsCertThumbprint":"4037FBF5F1F3014F99B5D6C7799E9B20E6871CB3","protectedSettings":"MIICWgYJK","publicSettings":{"foo":"bar"}}}]}</RuntimeSettings>
   </Plugin>
 </PluginSettings>
-<StatusUploadBlob statusBlobType="BlockBlob">https://yuezhatest.blob.core.windows.net/vhds/test-cs12.test-cs12.test-cs12.status?sr=b&amp;sp=rw&amp;se=9999-01-01&amp;sk=key1&amp;sv=2014-02-14&amp;sig=hfRh7gzUE7sUtYwke78IOlZOrTRCYvkec4hGZ9zZzXo%3D</StatusUploadBlob></Extensions>
+<StatusUploadBlob statusBlobType="BlockBlob">https://test.blob.core.windows.net/vhds/test-cs12.test-cs12.test-cs12.status?sr=b&amp;sp=rw&amp;se=9999-01-01&amp;sk=key1&amp;sv=2014-02-14&amp;sig=SOMESIG</StatusUploadBlob></Extensions>
```
Comment (narrieta) [inline tests/data/wire/ext_conf.xml]: why not leave the current signature, or change a few digits, to make it closer to real value

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
+mock
```
Comment (pgombar) [inline requirements.txt]: Why is this needed in `requirements.txt` if it's already part of `test-requirements.txt`?

File: azurelinuxagent/common/protocol/wire.py [hunk 3]
```diff
@@ -891,11 +891,11 @@ def get_current_handlers(self):
...
     def get_ext_conf(self):
         if self.ext_conf is None:
-            goal_state = self.get_goal_state()
-            if goal_state.ext_uri is None:
+            local_goal_state = self.get_goal_state()
+            if local_goal_state.ext_uri is None:
                 self.ext_conf = ExtensionsConfig(None)
             else:
-                local_file = EXT_CONF_FILE_NAME.format(goal_state.incarnation)
+                local_file = EXT_CONF_FILE_NAME.format(local_goal_state.incarnation)
```
Comment (pgombar) [inline azurelinuxagent/common/protocol/wire.py]: How does just renaming help with the variable not being referenced before assignment? Maybe I don't understand the issue.

Comment (vrdmr) [reply azurelinuxagent/common/protocol/wire.py]: tl;dr: This change is doing is making sure that the local `goal_state` is not conflicting with `self.goal_state`.

Initially, this issue was seen with the famous `update_goal_state` method (the logs have since gone away). The fix to rename the local variable `goal_state` in `update_goal_state` method was done in #1392.

Now it is complaining about the `get_ext_conf` method as per the stack trace, and I couldn't reproduce it but it looks like a repeat of the same case as above. (The above went away). It is one of the places where it complains (and other places with this issue are still to be determined)

One of the tracebacks example:
```
Exception processing extension handlers: local variable 'goal_state' referenced before assignment [
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/ga/exthandlers.py, line 253 in run>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/ga/exthandlers.py, line 514 in report_ext_handlers_status>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 192 in report_vm_status>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 1018 in upload_status_blob>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 894 in get_ext_conf>]
```

Other places where the traceback isn't showing up (And only impacting few hundred VMs repeatedly)
```
GuestAgentExtensionEvents
| where GAVersion startswith "WALinuxAgent-2.2.4" and Message contains "local variable 'goal_state' referenced before assignment"
| summarize count() by GAVersion, Message, Operation
| sort by count_ desc 

WALinuxAgent-2.2.42	Exception retrieving extension handlers: local variable 'goal_state' referenced before assignment	ExtensionProcessing	6954
WALinuxAgent-2.2.44	Exception retrieving extension handlers: local variable 'goal_state' referenced before assignment	ExtensionProcessing	4779
```

File: requirements.txt (hunk 2 above)
Comment (vrdmr) [reply requirements.txt]: Pycharm added it. Fixed it.

File: azurelinuxagent/common/protocol/wire.py (hunk 3 above)
Comment (narrieta) [reply azurelinuxagent/common/protocol/wire.py]: I can't see how this fixes #1026 either. I like renaming the local so that it is not named the same as the instance member, but let us not mark this PR as fixing #1026 (unless we have a good understanding that it does).

Comment (vrdmr) [reply azurelinuxagent/common/protocol/wire.py]: Fair. Updated the description.

Also, looking at the stack trace, it points to a dist-packages version of the azurelinuxagent which could be an old agent with this issue present. The newer agents are not adding the stack-trace with this issue and it could be an issue with mismatched GAVersion showing up in our telemetry.
